aten/src/ATen/cuda/tunable/GemmHipblaslt.h

```diff
@@ -0,0 +1,333 @@
+    int algo_index = GETINDEXFROMALGO(algo);
+    auto hipblaslt_gemm_op = [=](const ParamsT* params) -> TuningStatus {
+      auto opa = _hipblasOpFromChar(params->transa);
+      auto opb = _hipblasOpFromChar(params->transb);
+
+      TORCH_CHECK(transa_outer == opa && transb_outer == opb, "trans mismatch, shouldn't happen");
+
+      float alpha = static_cast<float>(params->alpha);
+      float beta = static_cast<float>(params->beta);
+
+      hipblasLtMatrixLayout_t mat_a, mat_b, mat_c;
+      hipblasLtMatmulDesc_t matmul;
+      if (opa == HIPBLAS_OP_N) {
+        TORCH_HIPBLASLT_CHECK(hipblasLtMatrixLayoutCreate(&mat_a, in_out_datatype, params->m, params->k, params->lda));
+      }
+      else {
+        TORCH_HIPBLASLT_CHECK(hipblasLtMatrixLayoutCreate(&mat_a, in_out_datatype, params->k, params->m, params->lda));
+      }
+      if (opb == HIPBLAS_OP_N) {
+        TORCH_HIPBLASLT_CHECK(hipblasLtMatrixLayoutCreate(&mat_b, in_out_datatype, params->k, params->n, params->ldb));
+      }
+      else {
+        TORCH_HIPBLASLT_CHECK(hipblasLtMatrixLayoutCreate(&mat_b, in_out_datatype, params->n, params->k, params->ldb));
+      }
+      TORCH_HIPBLASLT_CHECK(hipblasLtMatrixLayoutCreate(&mat_c, in_out_datatype, params->m, params->n, params->ldc));
+      TORCH_HIPBLASLT_CHECK(hipblasLtMatmulDescCreate(&matmul, COMPUTE_TYPE_32, DATA_TYPE_R_32));
+
+      TORCH_HIPBLASLT_CHECK(hipblasLtMatmulDescSetAttribute(
+            matmul, HIPBLASLT_MATMUL_DESC_TRANSA, &opa, sizeof(int32_t)));
+      TORCH_HIPBLASLT_CHECK(hipblasLtMatmulDescSetAttribute(
+            matmul, HIPBLASLT_MATMUL_DESC_TRANSB, &opb, sizeof(int32_t)));
+
+#if 0
+      // Deduce enable_bias from params
+      auto d_bias = GetBiasFromParams<T>(params);
+      bool enable_bias = d_bias != nullptr;
+
+      hipblasLtEpilogue_t epilogue;
+      switch (activation_type) {
+        case ActivationType::NONE:
+          epilogue = enable_bias ? HIPBLASLT_EPILOGUE_BIAS : HIPBLASLT_EPILOGUE_DEFAULT;
+          break;
+        case ActivationType::RELU:
+          epilogue = enable_bias ? HIPBLASLT_EPILOGUE_RELU_BIAS : HIPBLASLT_EPILOGUE_RELU;
+          break;
+        case ActivationType::GELU:
+          epilogue = enable_bias ? HIPBLASLT_EPILOGUE_GELU_BIAS : HIPBLASLT_EPILOGUE_GELU;
+          break;
+        default:
+          throw std::runtime_error("Unsupported activation type for HipBlasLtMatMul");
+      }
+      TORCH_HIPBLASLT_CHECK(hipblasLtMatmulDescSetAttribute(
+            matmul, HIPBLASLT_MATMUL_DESC_EPILOGUE, &epilogue, sizeof(epilogue)));
+
+      if (enable_bias) {
+        TORCH_HIPBLASLT_CHECK(hipblasLtMatmulDescSetAttribute(
+              matmul, HIPBLASLT_MATMUL_DESC_BIAS_POINTER, &d_bias, sizeof(void*)));
+      }
+#endif
+
+      size_t workspace_size = 2*128*1024*1024;
+
+      hipblasLtHandle_t op_handle;
+      TORCH_HIPBLASLT_CHECK(hipblasLtCreate(&op_handle));
+
+      //hipblasLtMatmulPreference_t preference;
```

Comment:
Clean up if not used? Is `HIPBLASLT_MATMUL_PREF_MAX_WORKSPACE_BYTES` needed to set?

Reply:
Done.